pkg/endpoint/policy.go

```diff
@@ -618,6 +619,58 @@ func (e *Endpoint) setRegenerateStateLocked(regenMetadata *regeneration.External
 	return regen
 }
 
+// UpdatePolicy updates the endpoint's policy.
+// If the endpoint's identity is in the set that needs regeneration, it will queue a regeneration
+// and wait for the result. If not, the endpoint's policy revision will be bumped to toRev without
+// a regeneration
+func (e *Endpoint) UpdatePolicy(idsToRegen *set.Set[identityPkg.NumericIdentity], fromRev, toRev uint64) {
+	// no deferred unlocks here, as we must
+	// release locks before regenerating
+
+	e.buildMutex.Lock() // buildMutex is required to update policy revision
+	if err := e.lockAlive(); err != nil {
+		e.buildMutex.Unlock()
+		return
+	}
+
+	unlock := func() {
+		e.unlock()
+		e.buildMutex.Unlock()
+	}
+
+	secID := e.getIdentity()
+	if secID == identityPkg.InvalidIdentity {
+		unlock()
+		return
+	}
+
+	// Does this identity have new policies, and thus needs
+	// to be regenerated?
+	if !idsToRegen.Has(secID) {
+		if e.policyRevision < fromRev {
```

Comment:
If the ID is affected, then we are *definitely* regenerating. If not, we need to see if it's safe to bump policy revisions. The comment is unclear, I'll clarify more.

Are you concerned about missing the error message in the must-regen case?

Reply:
>Are you concerned about missing the error message in the must-regen case?

Yep. In my mind this should never happen, but since the regeneration is the way to fix it, it doesn't matter much